src/lib/index/wb_cache.cpp

```diff
@@ -219,31 +242,33 @@ std::unique_ptr< CPContext > IndexWBCache::create_cp_context(cp_id_t cp_id) {
                                               m_free_blkid_list[cp_id_slot].get());
 }
 
-void IndexWBCache::do_flush_one_buf(IndexCPContext* cp_ctx, const IndexBufferPtr& buf, bool part_of_batch) {
-    LOGTRACEMOD(wbcache, "buf {}", buf->to_string());
-    buf->m_buf_state = index_buf_state_t::FLUSHING;
+void IndexWBCache::do_flush_one_buf(IndexCPContext* cp_ctx, IndexBufferPtr buf, bool part_of_batch) {
+    LOGTRACEMOD(wbcache, "cp {} buf {}", cp_ctx->id(), buf->to_string());
+    { buf->set_state(index_buf_state_t::FLUSHING); }
     m_vdev->async_write(r_cast< const char* >(buf->raw_buffer()), m_node_size, buf->m_blkid, part_of_batch)
```

Comment:
Instead of using IndexBuffer* , using IndexBufferPtr for safety and consistency.